upup/pkg/fi/cloudup/terraform/target_hcl2.go

```diff
@@ -132,23 +133,34 @@ func (t *TerraformTarget) writeProviders(buf *bytes.Buffer) {
 		Write(buf, 0, fmt.Sprintf("provider %q", providerName))
 	buf.WriteString("\n")
 
-	// Add the second provider definition for managed files
-	if t.filesProvider != nil {
+	// Add any additional provider definition for managed files
+	keys := keysForMap(t.TerraformWriter.Providers)
+	sort.Strings(keys)
+	for _, key := range keys {
+		filesProvider := t.TerraformWriter.Providers[key]
 		providerBody := map[string]string{}
 		providerBody["alias"] = "files"
```

Comment:
I don't think you can have multiple providers with the same alias.

Reply:
They can have the same alias as long as they are not the same provider type, which the `Providers` map should ensure

```tf
provider "null" {
  alias = "foo"
}

provider "random" {
  alias = "foo"
}

resource "null_resource" "bar" {
  provider = null.foo
}

resource "random_string" "bar" {
  provider = random.foo
  length = 1
}
```